### PR TITLE
added f1 score in log console training metrics and in results.txt (Issue #21454)

### PR DESCRIPTION
**Pull request by turcuciprian**
The purpose of this PR was to add the F1 score in the results.txt and also in the console info log when training

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

**WARNING ⚠️** this PR is very large, summary may not cover all changes.

### 🌟 Summary
Streamlines CI/CD, Docker builds, and documentation, while improving export/testing coverage and inference reliability. Adds new Vertex AI deployment guide and modernizes Docker images for faster, more modular builds. 🚀

### 📊 Key Changes
- CI/CD and Testing
  - Upgraded actions to checkout@v5 across workflows; refined conda job using uv; split pytest targets for clarity.
  - Temporarily disabled GPU CI job for stability.
  - Link checks use a direct lychee install; docs build pins click==8.2.1.
  - Publishing: actions/download-artifact@v5 and new SBOM generation via Anchore; safer PR title handling.
- Docker Build & Images
  - Major workflow overhaul: builds “base” images and their derivatives (runner, export, jupyter, cpu, python-export); unified push with versioned tags; login to NVIDIA NGC.
  - Environment checks/tests run against export variants when applicable; benchmarks aligned.
  - Core images updated:
    - Base GPU: PyTorch 2.8.0 CUDA 12.8; leaner apt; switch to opencv-python-headless.
    - New derivative images: Dockerfile-export, Dockerfile-python-export.
    - Dockerfile-python and Dockerfile-cpu simplified; CPU image now extends latest-python.
    - Jetson images cleaned with consistent installs and TF.js compatibility.
    - Runner uses JSON ENTRYPOINT; installs pytest-cov.
- Docs & Examples
  - New guide: Vertex AI Deployment with Docker; added tutorial videos across pages; content fixes and clarifications.
  - Updated mkdocs nav and banner promoting YOLO Vision.
  - Various doc code snippets corrected (SAM labels shape, SAHI imports, Streamlit section cleanup).
- Core Library Improvements
  - Version bump to 8.3.181.
  - Exporter: TorchScript supports dynamic; NCNN export avoids extra deps; IMX export deps pinned; YOLOE export fuses text embeddings.
  - Data/Transforms: LetterBox now configurable padding/interpolation; source list supports .csv; single-class label checks fixed.
  - Predict/Validate: Better batch/default handling for exported models.
  - SAM robustness: dtype/device-safe ops, interpolation fixes, improved mask/point handling; predictor returns tensors and supports feature-level inference.
- Tests
  - Expanded export test matrix: OpenVINO/ONNX dynamic, NCNN/MNN matrices; batch arguments verified; minor fixes.

### 🎯 Purpose & Impact
- Faster, more reliable CI/CD and publish pipelines ✅
- Modular Docker builds: quicker iteration, clearer specialization (e.g., export, python-export, runner, jupyter) 🐳
- More robust exports and broader format coverage improve deployment compatibility (ONNX/OpenVINO/NCNN/MNN/IMX) 📦
- Better runtime stability and accuracy for SAM and YOLOE models through dtype/device-safe changes and fused prompts 🧠
- Clearer docs and a practical Vertex AI deployment path help users operationalize YOLO11 quickly in the cloud ☁️
- Temporary GPU CI pause reduces flakiness while infrastructure stabilizes 🛠️

See the Ultralytics Docs for details.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @turcuciprian, thank you for submitting an `ultralytics/ultralytics` 🚀 PR! To ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don’t hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

_This is an automated response. An Ultralytics engineer will review your PR and assist you soon!_

**Comment by turcuciprian** [comment]
I have read the CLA Document and I sign the CLA

**Comment by turcuciprian** [comment]
Looking forward to feedback

**Comment by glenn-jocher** [comment]
Thanks @turcuciprian for this valuable contribution! Adding [F1-Score visibility to training outputs](https://www.ultralytics.com/glossary/f1-score) is a great enhancement that will help users better assess model performance, especially for imbalanced datasets where the harmonic mean of precision and recall provides more meaningful insights than accuracy alone. The PR looks well-structured and the automated CI checks should validate the implementation shortly.

**Comment by turcuciprian** [comment]
@glenn-jocher What should I do to get this reviewed, it's been 3 weeks and no progress was made on it on the reviewing side. Do I need to do anything specific?

**Comment by glenn-jocher** [comment]
Thanks for the nudge — please rebase on main, keep the diff scoped to logging/results-only (drop any fitness/behavior changes), and add a minimal test confirming F1 is emitted in the console output and results.txt; once CI is green, tag me and I’ll prioritize a review.

**Comment by turcuciprian** [comment]
> Thanks for the nudge — please rebase on main, keep the diff scoped to logging/results-only (drop any fitness/behavior changes), and add a minimal test confirming F1 is emitted in the console output and results.txt; once CI is green, tag me and I’ll prioritize a review.

done @glenn-jocher

**Comment by turcuciprian** [comment]
@glenn-jocher  done again (accidentally closed it previously)

File: ultralytics/utils/metrics.py [hunk 1]
```diff
@@ -875,12 +885,12 @@ def map(self) -> float:
...
     def mean_results(self) -> List[float]:
-        """Return mean of results, mp, mr, map50, map."""
-        return [self.mp, self.mr, self.map50, self.map]
+        """Return mean of results, mp, mr, map50,f1, map."""
+        return [self.mp, self.mr, self.map50,self.mf1, self.map]
 
     def class_result(self, i: int) -> Tuple[float, float, float, float]:
-        """Return class-aware result, p[i], r[i], ap50[i], ap[i]."""
-        return self.p[i], self.r[i], self.ap50[i], self.ap[i]
+        """Return class-aware result, p[i], r[i], ap50[i] ap[i]."""
```
Comment (onuralpszr) [inline ultralytics/utils/metrics.py:980]: f1 doc missing and also you take out "comma" by mistake I think :)

**Comment by onuralpszr** [comment]
> @glenn-jocher done again (accidentally closed it previously)

I hello, I see that you created a duplicate of old commits in PR while you doing rebase/merge to update your branch, it would be really awesome and great to that If you can fix that.  It makes easier for everyone who wants to read about PR but in a cleaner way and only see what changed. Thank you in advance.
